### PR TITLE
Add NetworkSwitcher in v5, use in dashboard

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [ ] Internal API changes: explain the internal logic changes

## How to test

- [ ] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

## Contributor NFT

Paste in your wallet address below and we will airdrop you a special NFT when your pull request is merged. 

```Address: ```

---

## PR-Codex overview
This PR introduces changes across multiple files to disable focus trapping and scroll blocking in various components.

### Detailed summary
- Disabled focus trapping and scroll blocking in `Modal`, `Drawer`, and other components
- Added `CustomChainRenderer` in `CustomConnectWallet`
- Updated `NetworkSelectorButton` to use `CustomChainRenderer`
- Implemented `mapStoredChainTov5Chain` function for mapping chains

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3549?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3549** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3397** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3397?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3392** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3392?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3 other dependent PRs ([#3393](https://github.com/thirdweb-dev/js/pull/3393) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3393?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3395](https://github.com/thirdweb-dev/js/pull/3395) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3395?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3396](https://github.com/thirdweb-dev/js/pull/3396) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3396?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Link from "next/link";
 import { useCallback, useMemo } from "react";
```
Comment (MananTank) [inline apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx:14]: Need to check something - On local the Network list was not scrollable when the NetworkSwitcher button is used in Deploy page. Need to check this in vercel preview to confirm

File: apps/dashboard/src/components/selects/CustomChainRenderer.tsx [hunk 2]
```diff
@@ -26,9 +32,13 @@ export const CustomChainRenderer = ({
   const setIsOpenNetworkConfigModal = useSetIsNetworkConfigModalOpen();
   const addRecentlyUsedChain = useAddRecentlyUsedChainId();
   const setEditChain = useSetEditChain();
+  const chainQuery = useQuery({
+    queryKey: ["getChain", chain.id],
+    queryFn: () => getChain(`${chain.id}`),
+  });
```
Comment (jnsdls) [inline apps/dashboard/src/components/selects/CustomChainRenderer.tsx:38]: can't to this it will make way too many requests

**Comment by MananTank** [comment]
Need to check something - On local the Network list was not scrollable when the NetworkSwitcher button is used in Deploy page. Need to check this in vercel preview to confirm

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx [hunk 3]
```diff
@@ -194,7 +201,8 @@ export function NetworkSelectorContent(props: NetworkSelectorContentProps) {
       ) {
         const recentChains = props.networkSelector.recentChainIds
           .map((id) => allChainsMap.get(id))
-          .filter((c) => c !== undefined);
+          .filter((c) => c !== undefined) as Chain[];
```
Comment (jnsdls) [inline packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx:204]: this should not be needed with typescript `5.5`

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx [hunk 4]
```diff
@@ -211,7 +219,8 @@ export function NetworkSelectorContent(props: NetworkSelectorContentProps) {
       ) {
         const popularChains = props.networkSelector.popularChainIds
           .map((id) => allChainsMap.get(id))
-          .filter((c) => c !== undefined);
+          .filter((c) => c !== undefined) as Chain[];
```
Comment (jnsdls) [inline packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx:222]: ditto

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/components/selects/CustomChainRenderer.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/components/selects/CustomChainRenderer.tsx:38]: V5 only renders chains that are visible

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx (hunk 3 above)
Comment (MananTank) [reply packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx:204]: nice

**Comment by MananTank** [comment]
DO NOT MERGE: need to fix scroll issue

**Comment by MananTank** [comment]
another issue - NetworkSelectorButton does not have all chains

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx (hunk 3 above)
Comment (MananTank) [reply packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx]: removed

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx (hunk 4 above)
Comment (MananTank) [reply packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx]: removed

File: apps/dashboard/src/components/selects/CustomChainRenderer.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/components/selects/CustomChainRenderer.tsx:38]: oh yeah, I can use use the chain object from useSupportedChains()

**Review by jnsdls — APPROVED** [review]

**Pull request merged**